.github/workflows/verify-build.yml

```diff
@@ -0,0 +1,79 @@
+# This workflow will perform a test whenever there
+# is some change in code done to ensure that the changes
+# are not buggy and we are getting the desired output.
+name: Test Build
+on: [push, pull_request, workflow_dispatch]
+env:
+  TEST_REPO: spring-petclinic
+  IMAGE_NAME: spring-petclinic
+
+jobs:
+  build:
+    name: Build image using Buildah
+    runs-on: ubuntu-latest
+    steps:
+
+      # Fetch name of the Forked Repository with Branch
+      # if workflow is triggered from pull request
+      - name: Fetch PR head repo and branch name
+        if: github.event_name == 'pull_request'
+        id: head_repo_info
+        run: |
+          HEAD_REPO_NAME=$(jq -r '.pull_request.head.repo.full_name' "$GITHUB_EVENT_PATH")
+          echo "PR head repo: $HEAD_REPO_NAME"
+          echo "::set-output name=repo::$HEAD_REPO_NAME"
+          echo "::set-output name=branch::$GITHUB_HEAD_REF"
+
+      # Extract branch name
+      - name: Extract branch name
+        if: github.event_name != 'pull_request'
+        shell: bash
+        run: echo "::set-output name=branch::${GITHUB_REF#refs/heads/}"
+
+      # Checkout buildah action github repository
+      - name: Checkout Buildah action
+        uses: actions/checkout@v2
+        with:
+          repository: ${{ steps.head_repo_info.outputs.repo }}
+          ref: ${{ steps.head_repo_info.outputs.branch }}
```

Comment:
If this is not a pull request, both outputs here will be empty. it means this will only work for the default branch in non-PR cases, which doesn't seem right.